src/storage-client/src/types/connections.rs

```diff
@@ -589,14 +589,18 @@ impl CsrConnection {
                 // computed for every request. This ensures that, if the tunnel
                 // fails and restarts at a new address, requests will start
                 // using the new tunnel address.
-                client_config = client_config.add_proxy(mz_ccsr::Proxy::custom(move |url| {
+
+                let remote_url = self.url.clone();
+                client_config = client_config.dynamic_url(move || {
                     let addr = ssh_tunnel.local_addr();
-                    let mut url = url.clone();
-                    url.set_host(Some(&addr.ip().to_string()))
-                        .expect("cannot fail");
-                    url.set_port(Some(addr.port())).expect("cannot fail");
-                    Some(url)
-                }));
+                    Url::parse(&format!(
+                        "{}://{}:{}",
+                        remote_url.scheme(),
+                        addr.ip(),
+                        addr.port()
+                    ))
+                    .unwrap()
```

Comment:
Why did this change? Isn't the original implementation sufficient? (I'm not sure whether it matters, but the original implementation was careful not to remove any `/base/path`.)

Reply:
hmm, you are right, we can just clone the url, let me go back to that!

Reply:
updated!